lisa/features/infiniband.py

```diff
@@ -411,16 +411,22 @@ def install_ofed(self) -> None:
         if not self._is_legacy_device():
             extra_params += " --skip-unsupported-devices-check"
 
-        node.execute(
-            f"{self.resource_disk_path}/{ofed_folder}/mlnxofedinstall "
-            f"--add-kernel-support {extra_params} "
-            f"--tmpdir {self.resource_disk_path}/tmp",
-            expected_exit_code=0,
-            expected_exit_code_failure_message="SetupRDMA: failed to install "
-            "OFED Drivers",
-            sudo=True,
-            timeout=1200,
-        )
+        try:
+            node.execute(
+                f"{self.resource_disk_path}/{ofed_folder}/mlnxofedinstall "
+                f"--add-kernel-support {extra_params} "
+                f"--tmpdir {self.resource_disk_path}/tmp",
+                expected_exit_code=0,
+                expected_exit_code_failure_message="SetupRDMA: failed to install "
+                "OFED Drivers",
+                sudo=True,
+                timeout=1200,
+            )
+        except AssertionError as e:
+            raise UnsupportedKernelException(
```

Comment:
it will catch all the exceptions?

Reply:
Yes, it will raise UnsupportedKernelException instead of AssertionError for all non-zero exit codes from the OFED driver install command.

Reply:
it misleads us if the real exception doesn't belong to `UnsupportedKernelException`

Reply:
@LiliDeng Which exception should I use for this scenario: prerequisite installation failed, not the test itself. `MissingPackagesException`? `NotMeetRequirementException`?

Reply:
@squirrelsc any comments?

Reply:
Is this ready to be merged?

Reply:
LGTM